content/docs/10.plugin-developer-guide/07.outputs.md

```diff
@@ -2,40 +2,98 @@
+Below is an example of an output object. It's a **dictionary** object that you can use to add any key-value pairs:
 
-```json5
-"outputs": { // map of key value with all the outputs you want to send
+```json
+"outputs": { // map/dicttionary of key-value pairs with outputs
     "my-key": "my-value",
-    "my-list": [1, 2, 3] // you can use  json type (bool, array, map, …)
+    "my-list": [1, 2, 3] // you can use  a variety of data types supported in a JSON request payload
 }
 ```
 
-Here is the representation of the metrics object, it's a list of JSON objects:
-```json5
-"metrics": [ // you can send multiple metrics at once
+Here is the representation of a `metrics` object. It's a **list of dictionaries**:
+```json
+"metrics": [ // you can send multiple metrics at once - each is a dictionary within a list
     {
         "name": "my-counter", // mandatory, the name of the metrics
-        "type": "counter", // mandatory, "counter" or "timer"
-        "value": 1.2, // mandatory (double), counter to add, or duration in seconds for timer
-        "tags":{ // optional list of tags that will expose internal details
-          "type":"read",
-          "location":"eu"
+        "type": "counter", // mandatory, "counter" or "timer" metric type
+        "value": 1.2, // mandatory (double), counter to measure number of objects/rows/items processed in a given task, or duration in seconds for the timer metric
+        "tags": { // optional list of tags that will expose internal details
+          "type": "read",
+          "location": "EU"
         }
     }
 ]
 ```
 
+## When to use metrics and when to use outputs?
+
+### Use cases for outputs
+
+Outputs are task-run artifacts. They are generated as a result of a given task. If you want to track task-run metadata across multiple executions of a flow, and this metadata is of an arbitrary data type (it might be a string, integer, ...), use `outputs` rather than `metrics`. 
+
+Examples of metadata you may want to track as `outputs`: 
+
+- the **number of rows** processed in a given task that you want to use in subsequent tasks validating the number of rows, 
+- the **accuracy score** of a trained ML model in order to compare this result (output artifact) across multiple workflow executions, 
+- other pieces of **metadata** you want to track across executions of a flow (e.g. a dataset name used within a Python ETL script).
+
+### Use cases for metrics
+
+Metrics are intended to track custom numeric (metric type: `counter`) or duration (metric type: `timer`) attributes that you may want to visualize across tasks. Metrics are typically expressed as `double` data type numerical values.
+
+Say, you are using the `EachParallel` task to process data across multiple partitions in parallel. You then want to visualize how many rows were processed in each partition and how long this process took. 
+
+Here is a simple flow demonstrating how you can accomplish that:
```

Comment:
```suggestion
Here is a flow demonstrating how you can accomplish that:
```

I always advise to not use words as 'simple', 'easy', ... in documentation as something that looks easy for you may not look easy for the ready which may be a little frustrated by the fact he didn't understand something that is simple ...